common/src/main/java/com/wynntils/core/config/ConfigManager.java

```diff
@@ -260,7 +261,7 @@ private List<ConfigHolder> getConfigOptions(Configurable parent) {
     }
 
     public List<ConfigHolder> getConfigHolders() {
-        return CONFIG_HOLDERS;
+        return Collections.unmodifiableList(CONFIG_HOLDERS);
```

Comment:
We only of of this in the wrong place and we will allocate a lot memory, won't we? Why do we need this anyways?

Reply:
The API from ConfigManager was not particularly streamlines for its actual purpose. I made this return a `Stream` instead which gets rid of the possibility to accidentally modify the list. I also fixed some other API issues that I ran into while doing it. Overall, it looks much smoother now.